atom/browser/browser_mac.mm

```diff
@@ -112,12 +114,16 @@
 void Browser::SetAppUserModelID(const base::string16& name) {
 }
 
-void Browser::SetUserActivity(
-    const std::string& type,
-    const base::DictionaryValue& user_info) {
+void Browser::SetUserActivity(const std::string& type,
+                              const base::DictionaryValue& user_info,
+                              mate::Arguments* args) {
+  std::string url_string;
+  args->GetNext(&url_string);
+
   [[AtomApplication sharedApplication]
       setCurrentActivity:base::SysUTF8ToNSString(type)
-            withUserInfo:DictionaryValueToNSDictionary(user_info)];
+            withUserInfo:DictionaryValueToNSDictionary(user_info)
+          withWebpageURL:net::NSURLWithGURL(GURL(url_string))];
```

Comment:
NSURL is now `nil` if `url` is invalid (i.e. not specified).
See [NSURLWithGURL](https://code.google.com/p/chromium/codesearch#chromium/src/net/base/mac/url_conversions.h&q=NSURLWithGURL&sq=package:chromium&type=cs&l=19)